polygeist/tools/cgeist/Lib/CGExpr.cc

```diff
@@ -2616,91 +2633,215 @@ ValueCategory MLIRScanner::EmitBinRem(const BinOpInfo &Info) {
   }
 }
 
+/// Casts index of subindex operation conditionally.
+static Optional<Value> castSubIndexOpIndex(OpBuilder &Builder, Location Loc,
+                                           ValueCategory Pointer,
+                                           ValueRange IdxList, bool IsSigned) {
+  if (Pointer.val.getType().isa<MemRefType>()) {
+    assert(IdxList.size() == 1 && "SubIndexOP accepts just an index");
+    return ValueCategory(IdxList.front(), false)
+        .IntCast(Builder, Loc, Builder.getIndexType(), IsSigned)
+        .val;
+  }
+  return llvm::None;
+}
+
+ValueCategory MLIRScanner::EmitCheckedInBoundsGEP(mlir::Type ElemTy,
```

Comment:
Should we modify the function name to indicate that it could generate SubIndexOp sometimes?

Reply:
Renamed to `EmitCheckedInBoundsPtrOffsetOp`.